### DISCUSSION TITLE
Azure DevOps Support

### DISCUSSION THREAD

**Discussion by aristosvo** [aquasecurity/chain-bench#95] [Ideas]
It would be great to have Azure DevOps Support. What are the things needed to be done to add a `scm-client`?

- [ ] Map the API's on the requirements
- [ ] Add tests
- [ ] Get a test environment to run acceptance tests against
- [ ] Any more...?

If we could make clear how to add more scm-clients I'd give it a try personally.

**Comment by pgoultiaev** [comment]
Great idea!

**Comment by mobyyou44** [comment]
Hi @aristosvo,
Sorry for the delay I miss this thread.
You can have a look at the new support for GitLab that we added recently: [Add Support for Gitlab](https://github.com/aquasecurity/chain-bench/pull/97)
and see [client_test.go](https://github.com/aquasecurity/chain-bench/blob/main/internal/scm-clients/github/client_test.go) for client test

Thanks for your help!

**Reply by aristosvo** [reply]
Thanks @mobyyou44!

For now I'm waiting for https://github.com/microsoft/azure-devops-go-api/issues/123 to be resolved, if that is available I can be more certain we'll have some proper output.

For client tests I'd probably need to generate a mock myself, let me see if that is an option.
